tests/components/local_ip/test.py

```diff
@@ -0,0 +1,39 @@
+"""Tests for the local_ip component."""
+import pytest
+
+from homeassistant.components.local_ip import DOMAIN
+from homeassistant.setup import async_setup_component
+from homeassistant.util import get_local_ip
+
+
+@pytest.fixture(name="config")
+def config_fixture():
+    """Create hass config fixture."""
+    return {DOMAIN: {"name": "test"}}
+
+
+async def test_basic_setup(hass, config):
+    """Test component setup creates entry from config."""
+    assert await async_setup_component(hass, DOMAIN, config)
+    await hass.async_block_till_done()
+    local_ip = get_local_ip()
```

Comment:
That matters inside pytest?

Reply:
We're using the same event loop here as for `hass` so we shouldn't block the loop as that could lead to unexpected behavior.

Reply:
When modifying this to `local_ip = await hass.async_add_executor_job(get_local_ip())`, I get a bunch of errors...

Any idea what I might be doing wrong?  I could swear I used add_executor_job the same way when writing the sensor (until I discovered the update-on-init param was a more intelligent way of doing what I wanted there)
```
INFO     homeassistant.loader:loader.py:205 Loaded test from custom_components.test
INFO     homeassistant.loader:loader.py:205 Loaded test_package from custom_components.test_package
INFO     homeassistant.loader:loader.py:205 Loaded local_ip from homeassistant.components.local_ip
INFO     homeassistant.setup:setup.py:153 Setting up local_ip
INFO     homeassistant.setup:setup.py:187 Setup of domain local_ip took 0.0 seconds.
DEBUG    homeassistant.core:core.py:570 Bus:Handling <Event component_loaded[L]: component=local_ip>
DEBUG    homeassistant.core:core.py:570 Bus:Handling <Event config_entry_discovered[L]>
ERROR    homeassistant.core:core.py:139 Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/workspaces/home-assistant/homeassistant/core.py", line 1207, in async_call
    raise ServiceNotFound(domain, service) from None
homeassistant.exceptions.ServiceNotFound: Unable to find service persistent_notification/dismiss
ERROR    homeassistant.core:core.py:139 Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/workspaces/home-assistant/homeassistant/core.py", line 1207, in async_call
    raise ServiceNotFound(domain, service) from None
homeassistant.exceptions.ServiceNotFound: Unable to find service persistent_notification/create
INFO     homeassistant.loader:loader.py:205 Loaded sensor from homeassistant.components.sensor
INFO     homeassistant.setup:setup.py:153 Setting up sensor
INFO     homeassistant.setup:setup.py:187 Setup of domain sensor took 0.0 seconds.
DEBUG    homeassistant.core:core.py:570 Bus:Handling <Event component_loaded[L]: component=sensor>
INFO     homeassistant.components.sensor:entity_platform.py:137 Setting up sensor.local_ip
DEBUG    homeassistant.core:core.py:570 Bus:Handling <Event state_changed[L]: entity_id=sensor.test, old_state=None, new_state=<state sensor.test=127.0.0.1; friendly_name=test @ 2019-12-31T09:48:36.713824+00:00>>
```

Reply:
Don't pass the return value of the function to `hass.async_add_executor_job`. Just pass the function. Ie don't call `get_local_ip`. It will be called later inside the thread pool.

Reply:
I copied that pattern from https://developers.home-assistant.io/docs/en/asyncio_working_with_async.html#calling-sync-functions-from-async

How would I capture the IP, then?

Reply:
```py
local_ip = await hass.async_add_executor_job(get_local_ip)
```